### PR TITLE
[stable/centrifugo] Update centrifugo to 2.4.0

### PR DESCRIPTION
**Pull request by yurymuski**
What this PR does / why we need it:
- update apiVersion: from extensions/v1beta1 to apps/v1 for support k8s > 1.16
- update centrifugo to 2.4.0
- updated labels following helm guideline
- added grpc support
- added namespace
- made deployment dependency on secret
- added possible app port configuration depending on values

#### Checklist

- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

### CODE REVIEW

**Comment by yurymuski** [comment]
@oleh-ozimok 
@cpanato

**Comment by yurymuski** [comment]
I've added myself to OWNERS and maintainers, can I do so? Or should I remove this part?

**Comment by yurymuski** [comment]
@cpanato ping. without apiVersion fix this chart does not work in kubernetes clusters newer than 1.16 version

**Comment by yurymuski** [comment]
+ added grpc support

File: stable/centrifugo/Chart.yaml [hunk 1]
```diff
@@ -11,4 +11,6 @@ home: https://github.com/centrifugal/centrifugo
 maintainers:
 - name: oleh-ozimok
   email: ozimok.oleg@gmail.com
+- name: yurymuski
+  email: muski.yury@gmail.com
```
Comment (cpanato) [inline stable/centrifugo/Chart.yaml]: please remove this part for now.

File: stable/centrifugo/OWNERS [hunk 2]
```diff
@@ -1,4 +1,8 @@
 approvers:
+- oleh-ozimok
 - cpanato
+- yurymuski
 reviewers:
+- oleh-ozimok
 - cpanato
+- yurymuski
```
Comment (cpanato) [inline stable/centrifugo/OWNERS]: here as well

**Comment by yurymuski** [comment]
@cpanato updates done

File: stable/centrifugo/templates/_helpers.tpl [hunk 3]
```diff
@@ -22,3 +30,22 @@ Create chart name and version as used by the chart label.
...
+app.kubernetes.io/managed-by: {{ .Release.Service }}
+{{- end -}}
+
+{{/*
+Selector labels
+*/}}
+{{- define "centrifugo.selectorLabels" -}}
+app.kubernetes.io/name: {{ include "centrifugo.name" . }}
+app.kubernetes.io/instance: {{ .Release.Name }}
+{{- end -}}
```
Comment (cpanato) [inline stable/centrifugo/templates/_helpers.tpl]: new line here

File: stable/centrifugo/templates/deployment.yaml [hunk 4]
```diff
@@ -1,14 +1,15 @@
-apiVersion: extensions/v1beta1
+apiVersion: apps/v1
```
Comment (cpanato) [inline stable/centrifugo/templates/deployment.yaml:1]: did you test with an existing deployment and then upgrade to this release? is it not going to break?

Comment (yurymuski) [reply stable/centrifugo/templates/deployment.yaml:1]: Yes, just tested on k8s 1.15 cluster, upgrade works well.
Note that current version does not work in 1.16+ clusters due to api deprecation
https://kubernetes.io/blog/2019/07/18/api-deprecations-in-1-16/

**Review by cpanato — APPROVED** [review]
/lgtm
/approve

**Pull request merged**